### PR TITLE
test(sbb-header): add visual tests

### PR DESCRIPTION
**Pull request by DavideMininni-Fincons**

### CODE REVIEW

File: src/elements/header/header-button/header-button.visual.spec.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+          await setup.withFixture(html`
+            <sbb-header-button icon-name="hamburger-menu-small" type="button"
+              >Menu</sbb-header-button
+            >
+          `);
+        }),
+      );
+
+      it(
+        `multiple=true ${state.name}`,
```
Comment (TomMenga) [inline src/elements/header/header-button/header-button.visual.spec.ts:22]: Does the multiple add any value?
I don't see any CSS rule in the header that affects multiple elements

File: src/elements/header/header-link/header-link.visual.spec.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        state.name,
+        state.with(async (setup) => {
+          await setup.withFixture(html`
+            <sbb-header-link icon-name="hamburger-menu-small">Menu</sbb-header-link>
+          `);
+        }),
+      );
+
+      it(
+        `multiple=true ${state.name}`,
```
Comment (TomMenga) [inline src/elements/header/header-link/header-link.visual.spec.ts:20]: Does the multiple add any value?
I don't see any CSS rule in the header that affects multiple elements

File: src/elements/header/header/header.visual.spec.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      it(
+        `expanded=${state.expanded} ${visualDiffDefault.name}`,
+        visualDiffDefault.with(async (setup) => {
+          await setup.withFixture(
+            html`
+              <style>
+                ${' .last-element {display: none;} '}
+                ${' @media screen and (width >= 840px}) { .last-element { display: block; } }'}
+                ${' @media screen and (width < 1023px) { .sbb-header-spacer { display: none; } }'}
+              </style>
```
Comment (TomMenga) [inline src/elements/header/header/header.visual.spec.ts:38]: In theory, we could add this global CSS in the `web-test-runner.config.ts` but I think it's better if we keep it here

**Review by TomMenga — APPROVED** [review]
LGTM 👍 
Some minor topics to discuss maybe

File: src/elements/header/header-button/header-button.visual.spec.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+import './header-button.js';
+
+describe(`sbb-header-button`, () => {
+  describeViewports({ viewports: ['zero', 'medium'] }, () => {
+    for (const state of visualDiffStandardStates) {
+      it(
+        state.name,
+        state.with(async (setup) => {
+          await setup.withFixture(html`
+            <sbb-header-button icon-name="hamburger-menu-small" type="button"
```
Comment (jeripeierSBB) [inline src/elements/header/header-button/header-button.visual.spec.ts:14]: type button is default
```suggestion
            <sbb-header-button icon-name="hamburger-menu-small"
```

File: src/elements/header/header-button/header-button.visual.spec.ts (hunk 1 above)
Comment (jeripeierSBB) [reply src/elements/header/header-button/header-button.visual.spec.ts:22]: I'd vote to remove them. multiple are tested with sbb-header?

File: src/elements/header/header/header.visual.spec.ts (hunk 3 above)
Comment (jeripeierSBB) [reply src/elements/header/header/header.visual.spec.ts:38]: I'd prefer to have it only in the test :)

File: src/elements/header/header/header.visual.spec.ts [hunk 5]
```diff
@@ -0,0 +1,65 @@
+import { html } from 'lit';
```
Comment (jeripeierSBB) [inline src/elements/header/header/header.visual.spec.ts:1]: Could we also test the scrolling stuff?

Comment (DavideMininni-Fincons) [reply src/elements/header/header/header.visual.spec.ts:1]: I gave it some try yesterday reducing the container height and using scrollTo/mockScrollTo, with no result :|

Comment (jeripeierSBB) [reply src/elements/header/header/header.visual.spec.ts:1]: I later try if I find a solution

**Review by jeripeierSBB — APPROVED** [review]
LGTM

**Pull request merged**